packages/rocketchat-api/server/v1/users.js

```diff
@@ -89,6 +89,44 @@ RocketChat.API.v1.addRoute('users.getPresence', { authRequired: true }, {
 	}
 });
 
+RocketChat.API.v1.addRoute('users.unreadRoomsCount', { authRequired: true }, {
+	get() {
+		const user = this.getUserFromParams();
+
+		let archived = false;
+
+		if (typeof this.queryParams.archived !== 'undefined') {
+			if (archived==='true') {
+				archived = true;
+			}
+		}
+
+		const unreadCount = RocketChat.models.Subscriptions.findUnreadRoomsByUserId(user._id, archived).count();
+
+		return RocketChat.API.v1.success({
+			count: unreadCount
+		});
+	}
+});
+
+RocketChat.API.v1.addRoute('users.subscriptions', { authRequired: true }, {
```

Comment:
Please consider adding support for the paging system. You can find their implementations in the various other api routes.

Reply:
Hi,
This route is returning subscriptions to the exact roomIds that are being sent in the request payload. Do you think there is a place for pagination here? It will cause an issue where you request information about a room but don't get it in response.

Reply:
Hi @graywolf336  - do you think paging is necessary here?